zcash_client_sqlite/src/lib.rs

```diff
@@ -2011,18 +2259,82 @@ mod tests {
+            .update_chain_tip(account.birthday().height())
+            .unwrap();
+
+        let current_addr = st
+            .wallet()
+            .get_last_generated_address(account.id(), None)
+            .unwrap();
         assert!(current_addr.is_some());
 
         let addr2 = st
             .wallet_mut()
             .get_next_available_address(account.id(), None)
-            .unwrap();
+            .unwrap()
+            .map(|(a, _)| a);
         assert!(addr2.is_some());
         assert_ne!(current_addr, addr2);
 
-        let addr2_cur = st.wallet().get_current_address(account.id()).unwrap();
+        let addr2_cur = st
+            .wallet()
+            .get_last_generated_address(account.id(), None)
+            .unwrap();
         assert_eq!(addr2, addr2_cur);
+
+        // Perform similar tests for shielded-only addresses. These should be timestamp-based; we
+        // will tick the clock between each generation.
+        use zcash_keys::keys::ReceiverRequirement::*;
+        #[cfg(feature = "orchard")]
+        let shielded_only_request = UnifiedAddressRequest::unsafe_new(Require, Require, Omit);
+        #[cfg(not(feature = "orchard"))]
+        let shielded_only_request = UnifiedAddressRequest::unsafe_new(Omit, Require, Omit);
+
+        let cur_shielded_only = st
+            .wallet()
+            .get_last_generated_address(account.id(), Some(shielded_only_request))
+            .unwrap();
+        assert!(cur_shielded_only.is_none());
+
+        let di_lower = st
+            .wallet()
+            .db()
+            .clock
+            .now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .expect("current time is valid")
+            .as_secs()
+            .saturating_add(MIN_SHIELDED_DIVERSIFIER_OFFSET);
+
+        let (shielded_only, di) = st
+            .wallet_mut()
+            .get_next_available_address(account.id(), Some(shielded_only_request))
+            .unwrap()
+            .expect("generated a shielded-only address");
+
+        // since not every Sapling diversifier index is valid, the resulting index will be bounded
+        // by the current time, but may not be equal to it
+        assert!(u128::from(di) >= u128::from(di_lower));
+
+        let cur_shielded_only = st
+            .wallet()
+            .get_last_generated_address(account.id(), Some(shielded_only_request))
+            .unwrap()
+            .expect("retrieved the last-generated shielded-only address");
+        assert_eq!(cur_shielded_only, shielded_only);
+
+        st.wallet_mut().db_mut().clock.tick(Duration::from_secs(10));
+
+        let (shielded_only_2, di_2) = st
+            .wallet_mut()
+            .get_next_available_address(account.id(), Some(shielded_only_request))
+            .unwrap()
+            .expect("generated a shielded-only address");
+        assert_ne!(shielded_only_2, shielded_only);
+        assert!(dbg!(u128::from(di_2)) >= dbg!(u128::from(di_lower) + 10));
```

Comment:
Were the `dbg!`s intended to be left in?

Reply:
No, good catch.